### DISCUSSION TITLE
Ubuntu Launcher 2.4.0

### DISCUSSION THREAD

**Discussion by jspw** [jspw/Ubuntu-Launcher#12] [General]
##### Fixes 👍🏼 
-  instant app disable, enable, uninstall and install listen and apps will be changed accordingly
- optimize performance

##### Feature
-  change option for shortcut apps on long pressing the app icon if not working well
-  Can change wallpaper
- user preference saved locally so after setup launcher will act faster

##### Suggested apk to download if you are confused 🥇 
-  [app-arm64-v8a-release.apk](https://github.com/jspw/Ubuntu-Launcher/releases/download/2.4.0/app-arm64-v8a-release.apk)

##### Any Question or Suggestion ? 
- Ask me in the release discussion

##### More detail about apk releases 💯 

```json

{
  "version": 2,
  "artifactType": {
    "type": "APK",
    "kind": "Directory"
  },
  "applicationId": "com.ubuntu.launcher",
  "variantName": "processReleaseResources",
  "elements": [
    {
      "type": "ONE_OF_MANY",
      "filters": [
        {
          "filterType": "ABI",
          "value": "x86_64"
        }
      ],
      "versionCode": 4009,
      "versionName": "2.4.0",
      "outputFile": "app-x86_64-release.apk"
    },
    {
      "type": "ONE_OF_MANY",
      "filters": [
        {
          "filterType": "ABI",
          "value": "arm64-v8a"
        }
      ],
      "versionCode": 2009,
      "versionName": "2.4.0",
      "outputFile": "app-arm64-v8a-release.apk"
    },
    {
      "type": "ONE_OF_MANY",
      "filters": [
        {
          "filterType": "ABI",
          "value": "armeabi-v7a"
        }
      ],
      "versionCode": 1009,
      "versionName": "2.4.0",
      "outputFile": "app-armeabi-v7a-release.apk"
    }
  ]
}

```

## What's Changed
* refactor code, change folder structure, optimize performance and add some features by @jspw in https://github.com/jspw/Ubuntu-Launcher/pull/10
* update app version and doc by @jspw in https://github.com/jspw/Ubuntu-Launcher/pull/11

**Full Changelog**: https://github.com/jspw/Ubuntu-Launcher/compare/2.3.1...2.4.0

<hr /><em>This discussion was created from the release <a href='https://github.com/jspw/Ubuntu-Launcher/releases/tag/2.4.0'>Ubuntu Launcher 2.4.0</a>.</em>

**Comment by dasilvasauro** [comment]
not working on android 12, "problem parsing the package" 😭

**Comment by nextgrwww** [comment]
Couldn't install on my Android 14 phone. Says package is not valid.

**Comment by ThatWhiteGuy364** [comment]
I'm attempting to install to my Moto G Play 2025, Android 14, I keep getting the error 'App not installed as it appears to not be a valid package'

**Comment by Coolgamer95-web** [comment]
Im trying to install this app but it say it's invalid
